### PR TITLE
Phase1

### PR DESCRIPTION
**Pull request by hector-valdivia**
Using vue2, vuex and inertia.js, I wasn't sure if it has to be a SPA, if that the case I will make it with vue3 and vite if you gave me de chance. Thanks

### CODE REVIEW

File: app/Http/Controllers/DashboardController.php [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+use App\Models\Entry;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Auth;
+use Inertia\Inertia;
+
+class DashboardController extends Controller
+{
+    public function index(Request $request){
+        $user = Auth::user();
```
Comment (A5hleyRich) [inline app/Http/Controllers/DashboardController.php:13]: You could grab the user from the request here:

```
$request->user();
```

File: app/Http/Requests/EntryRequest.php [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+class EntryRequest extends FormRequest
+{
+    /**
+     * Determine if the user is authorized to make this request.
+     *
+     * @return bool
+     */
+    public function authorize()
+    {
+        return Auth::check();
```
Comment (A5hleyRich) [inline app/Http/Requests/EntryRequest.php:17]: For update requests, this should validate that the current user owns the entry.

File: app/Http/Controllers/DashboardController.php [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+        return Inertia::render('Dashboard', [
+            'paginated' => Entry::where('user_id', $user->id)->orderBy('date', 'desc')->paginate(),
+            'total'     => Entry::select('amount')->where('user_id', $user->id)->sum('amount')
```
Comment (A5hleyRich) [inline app/Http/Controllers/DashboardController.php:17]: These could be cleaned up a little if the entries relationship is added to the user model:

```
$request->user()->entries()->orderBy('date', 'desc')->paginate()
$request->user()->entries()->sum('amount')
```

**Review by A5hleyRich — COMMENTED** [review]
Thanks for taking the time to complete phase 1 of the trial @hector-valdivia. This is a solid attempt. I've reviewed this PR like I would any other member of the team and left some feedback so you get a feel for our code review process.

Can you explain why you opted to use vuex in combination with inertia.js?

File: app/Http/Controllers/DashboardController.php (hunk 1 above)
Comment (hector-valdivia) [reply app/Http/Controllers/DashboardController.php:13]: Indeed I usually do that way `$request->user()`, which is more universal also if change the way of authentication, in summary was an inconsistency on my part

File: app/Http/Requests/EntryRequest.php (hunk 2 above)
Comment (hector-valdivia) [reply app/Http/Requests/EntryRequest.php:17]: I agree, I let myself be blinded, following only the instructions and I didn't think so much about the validation, can I send a fix branch or correct it in the same branch?

File: app/Http/Controllers/DashboardController.php (hunk 3 above)
Comment (hector-valdivia) [reply app/Http/Controllers/DashboardController.php:17]: Ok, some times I like to have the main model in hand to move if needed, but no problem changing it, and yes is more idiomatic and also is less pron to error

**Comment by hector-valdivia** [comment]
> Thanks for taking the time to complete phase 1 of the trial @hector-valdivia. This is a solid attempt. I've reviewed this PR like I would any other member of the team and left some feedback so you get a feel for our code review process.
> 
> Can you explain why you opted to use vuex in combination with inertia.js?

Yes, for the time given, auth method `laravel breeze` and best integration with vue  I obted for inertia.js, then used vuex to simplify the arrengement given to the components and avoid the emiting hell, also just where six mutation. 

Also could have used the `$page`  service, but in case the project have to turn into a SPA, vuex is safer

File: app/Http/Requests/EntryRequest.php (hunk 2 above)
Comment (A5hleyRich) [reply app/Http/Requests/EntryRequest.php:17]: No need to update the code.

**Comment by A5hleyRich** [comment]
Thanks for responding to the feedback. Is the phase 2 branch based of this PR? If so, could you merge this PR as it will make reviewing phase 2 easier.

**Comment by hector-valdivia** [comment]
> Thanks for responding to the feedback. Is the phase 2 branch based of this PR? If so, could you merge this PR as it will make reviewing phase 2 easier.

Yes, phase 2 branch is based on this PR and ok passing phase 1

**Pull request merged**